docs/proposals/secure-impersonation.md

```diff
@@ -0,0 +1,636 @@
+Alternatively, a ClusterRoleBinding on the reconciler's namespaced User name can grant access to all Namespaces.
+
+#### serviceAccount impersonation
+`spec.serviceAccountName` will remain supported, but impersonating Users is preferable because it reduces the attack surface within a Namespace.
+`ServiceAccounts` produce privileged token Secrets in the same Namespace often for use within `Pods`.
+It's possible to impersonate a `ServiceAccount` User name even when they don't exist, but it's possible to create the resource after the fact creating generated tokens without a change in the reconciler behavior.
+This doesn't happen with impersonated `Users` -- `Users` don't have any object representation. This prevents `Pods` in same Namespace as the `Kustomization`/`HelmRelease` from assuming the role of the privileged reconciler which would typically be an inappropriate privilege escalation.
+
+Impersonation of reconciler ServiceAccounts is done via API Headers that are permitted for the controller ServiceAccounts.
+The FluxUser Groups can still be added to requests for impersonated ServiceAccounts.
+The token-fetching impersonation method currently implemented within kustomize-controller and helm-controller should be removed or flagged off.
+This is a trade-off:
+- *Pro*: Simpler tenant configs -- possible to bind against and impersonate a ServiceAccount User name that "doesn't exist"/isn't represented by a Resource -- user not required to actually create an SA
+- *Pro*: No per-tenant crypto/random-number dependency due to SA-tokens
+- *Pro*: Controller SA's no longer require cluster-wide access to Secrets
+- *Pro*: Possible to configure controller to only have ServiceAccount impersonation via RBAC -- disable arbitrary User and Group Impersonation if controller code is untrusted (prevent use of `flux:user:*`)
+- **Con**: Single namespace installs of Flux controllers may depend on ClusterRoleBindings to flux-sa-impersonator which requires cluster-admin setup permissions.
+       Cannot(?) restrict ServiceAccount User impersonation to specific namespaces.
+       Compare to SA's and token Secrets which a Namespace admin has full access to *only within their Namespace*.
+       SA Token Impersonation allows single namespace installs to have less dependency on the cluster-admin (as long as Flux CRD's are already registered) while still keeping reconciler specific users.
+       For this reason, it may be nice to keep `--sa-token-impersonation` as a controller flag and support it with the single-namespace installation option, despite the faults /w using SA's.
+       It's possible to do some heuristic fallback here, but it could complicate things.
+
+If SA Token impersonation is enabled, impersonated FluxUser Groups are omitted from the request.
+ServiceAccounts still have a Group that you can rolebind against: `system:serviceaccounts:{{namespace}}`.
+
+#### defaulting
+Behavior must be clarified when both `user` and `serviceAccountName` are provided.
```

Comment:
If we don't need different usernames per namespace and remove the ability to impersonate SAs but use the pod SA - then maybe this isn't even needed?